package com.facebook.ads.redexgen.core;

import com.facebook.ads.sync.SyncModifiableBundle;
import com.vungle.ads.internal.protos.Sdk;
import f6.q;
import java.net.URLEncoder;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Iterator;
import java.util.List;
import java.util.Locale;
import java.util.Map;
import java.util.Set;
import java.util.concurrent.CountDownLatch;
import java.util.concurrent.atomic.AtomicReference;
import l3.a;
import n0.w;
import org.json.JSONException;
import org.json.JSONObject;
import org.json.JSONTokener;
import qb.d;
import zi.c;

/* JADX INFO: renamed from: com.facebook.ads.redexgen.X.2U, reason: invalid class name */
/* JADX INFO: loaded from: assets/audience_network/classes2.dex */
public final class C2U implements CY, InterfaceC2881gP {
    public static byte[] A0A;
    public static String[] A0B = {"IFN7jxPm5iUVHezzrmxS", "S0eu2eNSRgR", "zAFRaPaLH2briFok", "kmV", "0tQTfhSJ7", "pW9DdJxYDevwUXMYU", "23M9j", "tE9Nw4uJT9AR2sagm"};
    public final T8 A00;
    public final TP A01;
    public final InterfaceC2851fv A02;
    public final InterfaceC2882gQ A03;
    public final C2890gY A04;
    public final InterfaceC2891gZ A05;
    public final String A06;
    public final Map<EnumC2877gL, C1878Cb> A09 = new HashMap();
    public final Map<EnumC2877gL, AbstractC16412d> A08 = new HashMap();
    public final List<InterfaceC2895gd> A07 = new ArrayList();

    public static String A01(int i10, int i11, int i12) {
        byte[] bArrCopyOfRange = Arrays.copyOfRange(A0A, i10, i10 + i11);
        for (int i13 = 0; i13 < bArrCopyOfRange.length; i13++) {
            bArrCopyOfRange[i13] = (byte) ((bArrCopyOfRange[i13] - i12) - 21);
        }
        return new String(bArrCopyOfRange);
    }

    /* JADX WARN: Failed to parse debug info
    java.lang.ArrayIndexOutOfBoundsException: Index 13 out of bounds for length 13
    	at jadx.plugins.input.dex.sections.debuginfo.DebugInfoParser.startVar(DebugInfoParser.java:203)
    	at jadx.plugins.input.dex.sections.debuginfo.DebugInfoParser.process(DebugInfoParser.java:125)
    	at jadx.plugins.input.dex.sections.DexCodeReader.getDebugInfo(DexCodeReader.java:122)
    	at jadx.core.dex.nodes.MethodNode.getDebugInfo(MethodNode.java:656)
    	at jadx.core.dex.visitors.debuginfo.DebugInfoAttachVisitor.visit(DebugInfoAttachVisitor.java:38)
     */
    private void A04() throws Throwable {
        HashMap map = new HashMap();
        HashMap map2 = new HashMap();
        HashMap map3 = new HashMap();
        HashMap map4 = new HashMap();
        synchronized (this) {
            for (C1878Cb c1878Cb : this.A09.values()) {
                if (c1878Cb.AAe()) {
                    map.put(c1878Cb, EnumC2898gg.A05);
                } else {
                    map.put(c1878Cb, EnumC2898gg.A04);
                }
            }
            new AtomicReference();
            new AtomicReference();
            Iterator<AbstractC16412d> it = this.A08.values().iterator();
            while (it.hasNext()) {
                it.next();
                InterfaceC2876gK interfaceC2876gK = null;
                if (interfaceC2876gK.AAe()) {
                    throw new NullPointerException(A01(349, 20, 51));
                }
            }
            Iterator<InterfaceC2895gd> it2 = this.A07.iterator();
            if (it2.hasNext()) {
                it2.next();
                throw new NullPointerException(A01(w.c.f115839v, 22, 120));
            }
        }
        CountDownLatch countDownLatch = new CountDownLatch(1);
        AtomicReference atomicReference = new AtomicReference();
        AtomicReference atomicReference2 = new AtomicReference();
        JSONObject jSONObjectA03 = A03(map, map2, map3, map4);
        String.format(Locale.US, A01(188, 33, 57), this.A06, jSONObjectA03.toString(2));
        this.A02.AGy(this.A06, (A01(327, 8, 121) + URLEncoder.encode(jSONObjectA03.toString())).getBytes(), new CN(this, atomicReference, atomicReference2, countDownLatch));
        while (countDownLatch.getCount() > 0) {
            try {
                countDownLatch.await();
            } catch (InterruptedException unused) {
            }
        }
        synchronized (this) {
            if (atomicReference2.get() != null) {
                throw ((Throwable) atomicReference2.get());
            }
            Set<InterfaceC2894gc> setA02 = A02((JSONObject) atomicReference.get());
            HashMap map5 = new HashMap();
            HashMap map6 = new HashMap();
            Iterator<InterfaceC2894gc> it3 = setA02.iterator();
            while (it3.hasNext()) {
                it3.next().A4D(map5, map6);
            }
            this.A03.A6b(A00((JSONObject) atomicReference.get()));
            Iterator<InterfaceC2895gd> it4 = this.A07.iterator();
            if (it4.hasNext()) {
                it4.next();
                throw new NullPointerException(A01(401, 23, 20));
            }
        }
    }

    public static void A05() {
        A0A = new byte[]{102, -81, -71, 102, -72, -85, -71, -70, -72, -81, -87, -70, -85, -86, 116, 102, -103, -79, -81, -74, -74, -81, -76, -83, 102, -71, -65, -76, -87, -43, 4, 4, -76, -3, 2, -76, -10, -11, -9, -1, -5, 6, 3, 9, 2, -8, a.f103452q7, -67, -37, q.B, q.B, -23, -18, -102, -35, -20, -33, -37, -18, -33, -102, -19, -33, -20, -16, -33, -20, -102, -36, -17, q.B, -34, -26, -33, -102, -15, -29, -18, -30, -102, q.B, -23, q.B, -89, -19, -33, -20, -16, -33, -20, -102, -23, -15, q.B, -33, -34, -102, -36, -17, q.B, -34, -26, -33, -102, a.f103460r7, -66, 123, -104, -85, -104, 87, -89, -87, -90, -102, -100, -86, -86, -96, -91, -98, 87, -90, -89, -85, -96, -90, -91, 87, -118, -81, -73, -94, -83, -86, -91, 97, -77, -90, -89, -77, -90, -76, -87, 97, -75, -86, -82, -90, 123, 97, 102, -91, -34, -11, 7, -80, 3, -11, 2, 6, -11, 2, -67, -1, 7, -2, -11, -12, -80, q.f83622z, 5, -2, -12, -4, -11, -80, -13, 2, -11, -15, 4, -11, -12, a.f103502w7, -80, -75, 3, -95, a.f103484u7, -68, -79, -74, a.f103436o7, -67, -68, -73, -56, -81, a.f103452q7, -73, -67, -68, 110, a.f103436o7, -77, -65, a.f103460r7, -77, a.f103444p7, a.f103452q7, 110, a.f103452q7, -67, 110, 115, a.f103444p7, -120, 88, 115, a.f103444p7, -34, 4, -7, -18, -13, -3, -6, -7, -12, 5, -20, -1, -12, -6, -7, -85, -3, -16, -2, -5, -6, -7, -2, -16, a.f103468s7, -107, -80, -2, -43, -5, -16, -27, -22, -12, -15, -16, -21, -4, -29, -10, -21, -15, -16, -94, -12, -9, -16, -94, q.B, -29, -21, -18, -25, -26, -67, -94, q.B, -15, -12, -27, -21, -16, -23, -94, -21, -16, -94, -89, -26, -94, -11, -25, -27, -15, -16, -26, -11, -28, -9, -16, -26, -18, -25, -11, a.f103460r7, a.A7, a.f103529z7, -44, a.f103468s7, a.f103428n7, -44, 124, 121, -116, 121, -90, -87, -82, -89, -91, -78, -80, -78, -87, -82, -76, -2, -17, 7, -6, -3, -17, q.f83622z, a.f103511x7, -116, 127, -128, -116, 127, -115, -126, -10, -23, -11, -7, -23, -9, -8, -70, -83, -71, -67, -79, -70, -83, -69, -101, a.f103444p7, -74, -85, -119, -74, -84, -102, -83, -69, -83, -68, -69, -82, -68, -71, -72, -73, -68, -82, 0, q.f83622z, -1, 3, q.f83622z, -1, -36, 4, -5, q.f83622z, -15, a.A7, 2, -5, -15, -7, q.f83622z, -48, -1, q.f83622z, -18, 1, q.f83622z, -15, -100, -94, -105, -116, -111, -101, -104, -105, -110, -93, -118, -99, -110, -104, -105, 111, -110, -105, -110, -100, -111, -114, -115, 0, 6, -5, -16, -11, -1, -4, -5, -10, 7, -18, 1, -10, -4, -5, -32, 1, -18, -1, 1, q.f83622z, -15, -11, -30, -13, q.B, -26, -11, -32, -13, -26, -25, -13, -26, -12, -23, -32, -12};
    }

    /* JADX WARN: Failed to parse debug info
    java.lang.ArrayIndexOutOfBoundsException
     */
    private synchronized void A08(Throwable th2) {
        Iterator<InterfaceC2895gd> it = this.A07.iterator();
        if (it.hasNext()) {
            it.next();
            new HashMap();
            new HashMap();
            throw new NullPointerException(A01(401, 23, 20));
        }
    }

    /* JADX WARN: Failed to parse debug info
    java.lang.ArrayIndexOutOfBoundsException
     */
    @Override // com.facebook.ads.redexgen.core.CY
    public final synchronized InterfaceC2876gK A5U(EnumC2877gL enumC2877gL) {
        if (enumC2877gL.A03() != EnumC2879gN.A04) {
            throw new IllegalArgumentException(A01(47, 59, 101));
        }
        if (this.A09.containsKey(enumC2877gL)) {
            return this.A09.get(enumC2877gL);
        }
        C1878Cb c1878Cb = new C1878Cb(enumC2877gL);
        this.A09.put(enumC2877gL, c1878Cb);
        Iterator<InterfaceC2895gd> it = this.A07.iterator();
        if (it.hasNext()) {
            it.next();
            throw new NullPointerException(A01(377, 24, 120));
        }
        String.format(Locale.US, A01(153, 35, 123), enumC2877gL);
        return c1878Cb;
    }

    static {
        A05();
    }

    public C2U(T8 t10, TP tp2, InterfaceC2851fv interfaceC2851fv, String str, InterfaceC2891gZ interfaceC2891gZ, C2890gY c2890gY, InterfaceC2880gO interfaceC2880gO) {
        this.A00 = t10;
        this.A01 = tp2;
        this.A02 = interfaceC2851fv;
        this.A06 = str;
        this.A05 = interfaceC2891gZ;
        this.A04 = c2890gY;
        this.A03 = interfaceC2880gO.A5E(this);
    }

    public static int A00(JSONObject jSONObject) throws JSONException {
        int time = jSONObject.getJSONObject(A01(335, 7, 5)).getInt(A01(446, 16, 108));
        if (time > 0) {
            return time;
        }
        throw new JSONException(String.format(Locale.US, A01(129, 24, 44), Integer.valueOf(time)));
    }

    private Set<InterfaceC2894gc> A02(JSONObject jSONObject) throws JSONException {
        HashSet hashSet = new HashSet();
        JSONObject jSONObject2 = jSONObject.getJSONObject(A01(369, 8, 52));
        JSONObject data = jSONObject.getJSONObject(A01(298, 7, 109));
        Iterator<C1878Cb> it = this.A09.values().iterator();
        while (true) {
            boolean zHasNext = it.hasNext();
            String[] strArr = A0B;
            if (strArr[6].length() == strArr[0].length()) {
                throw new RuntimeException();
            }
            String[] strArr2 = A0B;
            strArr2[1] = "GiFNWD8Qhge";
            strArr2[3] = "8JT";
            if (zHasNext) {
                final C1878Cb next = it.next();
                switch (C2893gb.A01[EnumC2899gh.A00(jSONObject2.getString(next.A8F().A04())).ordinal()]) {
                    case 1:
                        final JSONObject fingerprint = data.getJSONObject(next.A8F().A04()).getJSONObject(A01(312, 4, 3));
                        final JSONObject responseObject = data.getJSONObject(next.A8F().A04()).optJSONObject(A01(316, 11, 43));
                        hashSet.add(new CL(next, fingerprint, responseObject) { // from class: com.facebook.ads.redexgen.X.2X
                            public final JSONObject A00;
                            public final JSONObject A01;

                            {
                                EnumC2899gh enumC2899gh = EnumC2899gh.A03;
                                this.A00 = fingerprint;
                                this.A01 = responseObject;
                            }

                            @Override // com.facebook.ads.redexgen.core.CL, com.facebook.ads.redexgen.core.InterfaceC2894gc
                            public final void A4D(Map<InterfaceC2876gK, EnumC2899gh> map, Map<SyncModifiableBundle, EnumC2886gU> map2) {
                                super.A00.A03(this.A00, this.A01);
                                super.A4D(map, map2);
                            }
                        });
                        break;
                    case 2:
                        hashSet.add(new CL(next) { // from class: com.facebook.ads.redexgen.X.2W
                            {
                                EnumC2899gh enumC2899gh = EnumC2899gh.A04;
                            }
                        });
                        break;
                    default:
                        throw new AssertionError();
                }
            } else {
                Iterator<AbstractC16412d> it2 = this.A08.values().iterator();
                while (it2.hasNext()) {
                    it2.next();
                    final AbstractC16412d abstractC16412d = null;
                    switch (C2893gb.A00[EnumC2886gU.A00(jSONObject2.getString(abstractC16412d.A8F().A04())).ordinal()]) {
                        case 1:
                            hashSet.add(new CM(abstractC16412d) { // from class: com.facebook.ads.redexgen.X.2b
                                {
                                    EnumC2886gU enumC2886gU = EnumC2886gU.A03;
                                }
                            });
                            break;
                        case 2:
                            hashSet.add(new CM(abstractC16412d) { // from class: com.facebook.ads.redexgen.X.2V
                                public static byte[] A00;

                                static {
                                    A01();
                                }

                                public static String A00(int i10, int i11, int i12) {
                                    byte[] bArrCopyOfRange = Arrays.copyOfRange(A00, i10, i10 + i11);
                                    for (int i13 = 0; i13 < bArrCopyOfRange.length; i13++) {
                                        bArrCopyOfRange[i13] = (byte) ((bArrCopyOfRange[i13] - i12) - 102);
                                    }
                                    return new String(bArrCopyOfRange);
                                }

                                public static void A01() {
                                    A00 = new byte[]{48, 34, 49, c.f161639q, 34, 46, 50, 38, 47, 34, 48, c.f161640r, 54, 43, 32};
                                }

                                {
                                    EnumC2886gU enumC2886gU = EnumC2886gU.A04;
                                }

                                @Override // com.facebook.ads.redexgen.core.CM, com.facebook.ads.redexgen.core.InterfaceC2894gc
                                public final void A4D(Map<InterfaceC2876gK, EnumC2899gh> map, Map<SyncModifiableBundle, EnumC2886gU> map2) {
                                    throw new NullPointerException(A00(0, 15, 87));
                                }
                            });
                            break;
                        default:
                            throw new AssertionError();
                    }
                }
                return hashSet;
            }
        }
    }

    private JSONObject A03(Map<InterfaceC2876gK, EnumC2898gg> map, Map<SyncModifiableBundle, EnumC2885gT> map2, Map<SyncModifiableBundle, JSONObject> clientBundleData, Map<SyncModifiableBundle, JSONObject> clientBundleFingerprint) throws JSONException {
        String strA01;
        JSONObject syncRequest = new JSONObject();
        for (Map.Entry<InterfaceC2876gK, EnumC2898gg> entry : map.entrySet()) {
            syncRequest.put(entry.getKey().A8F().A04(), entry.getValue().A03());
        }
        for (Map.Entry<SyncModifiableBundle, EnumC2885gT> entry2 : map2.entrySet()) {
            entry2.getKey();
            InterfaceC2876gK interfaceC2876gK = null;
            String strA04 = interfaceC2876gK.A8F().A04();
            String strA03 = entry2.getValue().A03();
            String[] strArr = A0B;
            if (strArr[1].length() == strArr[3].length()) {
                throw new RuntimeException();
            }
            A0B[4] = "GjL9vB6wh";
            syncRequest.put(strA04, strA03);
        }
        JSONObject jSONObject = new JSONObject();
        Iterator<Map.Entry<InterfaceC2876gK, EnumC2898gg>> it = map.entrySet().iterator();
        while (true) {
            boolean zHasNext = it.hasNext();
            strA01 = A01(316, 11, 43);
            if (!zHasNext) {
                break;
            }
            Map.Entry<InterfaceC2876gK, EnumC2898gg> next = it.next();
            if (next.getValue() == EnumC2898gg.A05) {
                JSONObject updateData = new JSONObject();
                InterfaceC2876gK key = next.getKey();
                jSONObject.put(key.A8F().A04(), updateData);
                if (next.getKey().A8F().A05()) {
                    JSONObject request = key.A88();
                    updateData.put(strA01, request);
                } else {
                    updateData.put(strA01, JSONObject.NULL);
                }
                EnumC2877gL enumC2877gLA8F = key.A8F();
                String[] strArr2 = A0B;
                if (strArr2[7].length() != strArr2[2].length()) {
                    String[] strArr3 = A0B;
                    strArr3[1] = "cfjTgROdu8b";
                    strArr3[3] = "qIs";
                    jSONObject.put(enumC2877gLA8F.A04(), updateData);
                } else {
                    A0B[4] = "vaTQWmMwm";
                    jSONObject.put(enumC2877gLA8F.A04(), updateData);
                }
            }
        }
        for (Map.Entry<SyncModifiableBundle, EnumC2885gT> entry3 : map2.entrySet()) {
            JSONObject updateData2 = new JSONObject();
            InterfaceC2876gK key2 = entry3.getKey();
            JSONObject request2 = clientBundleFingerprint.get(entry3.getKey());
            updateData2.put(strA01, request2);
            if (entry3.getValue() == EnumC2885gT.A05) {
                String strA02 = A01(312, 4, 3);
                JSONObject request3 = clientBundleData.get(key2);
                updateData2.put(strA02, request3);
                jSONObject.put(key2.A8F().A04(), updateData2);
            } else {
                jSONObject.put(key2.A8F().A04(), updateData2);
            }
        }
        JSONObject jSONObject2 = new JSONObject();
        for (Map.Entry<String, String> entry4 : this.A05.A7z().entrySet()) {
            if (entry4.getValue() != null) {
                jSONObject2.put(entry4.getKey(), entry4.getValue());
            }
        }
        JSONObject jSONObject3 = new JSONObject();
        jSONObject3.put(A01(342, 7, 111), syncRequest);
        jSONObject3.put(A01(298, 7, 109), jSONObject);
        jSONObject3.put(A01(305, 7, 75), jSONObject2);
        return jSONObject3;
    }

    public static void A06(String str, AtomicReference<JSONObject> atomicReference, AtomicReference<Throwable> atomicReference2) {
        String.format(Locale.US, A01(Sdk.SDKError.Reason.AD_LOAD_FAIL_RETRY_AFTER_VALUE, 28, 118), str);
        try {
            atomicReference.set((JSONObject) new JSONTokener(str).nextValue());
        } catch (ClassCastException | JSONException e10) {
            atomicReference2.set(e10);
        }
    }

    @Override // com.facebook.ads.redexgen.core.CY, com.facebook.ads.redexgen.core.TE
    public final void A6d() {
        this.A03.A6c();
    }

    @Override // com.facebook.ads.redexgen.core.InterfaceC2881gP
    public final void AIo() {
        if (this.A01.AAh()) {
            String str = A01(106, 23, 34) + this.A01.A7k().A07() + A01(0, 29, 49);
            A08(new C2897gf());
            return;
        }
        try {
            if (!C2350Up.A27(this.A00) || this.A00.A04().AAU()) {
                A04();
                return;
            }
            throw new IllegalStateException(A01(29, 18, 127));
        } catch (Throwable th2) {
            String.format(Locale.US, A01(d.f122118j, 49, 109), Integer.valueOf(this.A04.A01()));
            A08(th2);
            this.A03.A6b(this.A04.A01());
        }
    }
}
